### DISCUSSION TITLE
Local Host cannot be reached

### DISCUSSION THREAD

**Discussion by sophielamb** [thegetty/quire#39] [Get Help] [answered]
Hi again,
I hope I don't become a regular on this sub-board. This is probably a simple problem, however I spent a couple of days googling which didn't turn up a solution.

All of the software is now set-up and is open in the text editor (I'm a Windows user). However, I cannot open the local host to preview the project - it says "localhost refused to connect". I have tried the address in Opera, IE and Chrome, in case it was to do with browser settings.

Is this a connection problem? Could the files be in the wrong directories and therefore not found? Again, it's possibly something simple but I'm not sure where the problem might be. 

I'm really grateful you've set up this board, and am really looking forward to getting going with Quire! Thank you

**Comment by geealbers** [comment, marked as answer]
Hi @sophielamb, this may be a bug in our latest release. In your project folder, look inside the `themes/default` folder and see if there is a `node_modules` folder in there. If there's not, back in PowerShell run `quire install` and then when that's complete try `quire preview` again. Let me know if that fixes it. If not, stop the preview by running `quire stop` or typing Control-C in PowerShell, and instead run `quire preview --verbose`. This will give you a lot more output text and if you can share that here, it will help us diagnose the problem further.

And was this a new project you started with `quire new`, or did you clone an existing project from somewhere else?

**Reply by sophielamb** [reply]
Hi @geealbers, I created a new quire project as I wanted to get to know the basics.

I looked in `themes/default` and the folder isn't there. I ran `quire install` which threw up the error 
`(node:16512) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'config.yml'` 

I then ran `npm install --global @thegetty/quire-cli` since that's the way to update to the latest version. Afterwards I tried `quire install` again but that resulted in the previous error. It advised using `node --trace-warnings` to show where the warning was created.

        PS C:\Users\sophi> node --trace-warnings ...
        internal/modules/cjs/loader.js:883
          throw err;
          ^
        
        Error: Cannot find module 'C:\Users\sophi\...'
        ←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)←[39m
        ←[90m    at Function.Module._load (internal/modules/cjs/loader.js:725:27)←[39m
        ←[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)←[39m
        ←[90m    at internal/main/run_main_module.js:17:47←[39m {
          code: ←[32m'MODULE_NOT_FOUND'←[39m,
          requireStack: []
        }


I'm also not certain where all the Quire files are located on my computer, which probably doesn't help. The `nodejs` file is in `C:/Program Files`, Prince is in `C:/Program Files (x86)`, the `practice-project` I created was put into `C:/Users/sophi` and there was an automatically created `.node_repl_history` file in there too.

**Reply by geealbers** [reply]
Hi @sophielamb, I'm very sorry you're having this trouble, we'll get it sorted though!

Open PowerShell and the prompt should say `PS C:\Users\sophi>` before running quire commands, you'll need to move into your Quire project from here. Type `cd practice-project` and hit enter. That will "change directory" into your project named "practice-project". The prompt should then say `PS C:\Users\sophi\practice-project>`, if it does you should be able to run `quire install`. Ignore any warnings or errors you may see for now, and when that process stops, try `quire preview --verbose` and check http://localhost:1313/ in your browser.  If you're still not seeing the preview properly, I'd suggest we find a time to get on a screen-sharing call where we can walk through it together.

**Reply by sophielamb** [reply]
Hi @geealbers - your patience is wonderful, I'm so grateful that you're helping walk me through all of this.

Okay, not navigating to the right folder was a rookie error - once I was inside "practice-project", I could run ``quire install``. I then commanded ``quire preview --verbose`` which worked fine, although it did bring up a list of errors related to not finding some of the 'node-sass' modules. 

However - http://localhost:1313 now works mostly fine, which is fantastic progress, although I noticed that the navbar menu button does not work. The console inspect tells me that the toggle on click even is not defined, and that the node-sass module failed to build.

If it is possible, I'd very much welcome a call in order to talk through step by step.

**Reply by geealbers** [reply]
Just to wrap this up for anyone looking at this ...

Some of the errors @sophielamb was seeing in PowerShell were related to **node-sass** and **node-gyp**. These showed up when she was running `quire preview --verbose` and she was able to preview the project in her browser but found the javascript functions weren't working. Namely, the sidebar menu wouldn't open.

These errors are related to dependencies that are installed in a project when starting a new project or running `quire install`. There are some of these dependencies that require an extra level of processing, and it's this process piece that was breaking.

**SOLUTION:**

1. Open file explorer and delete the `themes/default/node_modules` folder from your project
2. Open PowerShell in Administrator mode and run `npm install --g --production windows-build-tools`
3. `cd` into your project and run `quire install` (which rebuilds the `themes/default/node_modules` folder)
4. Test by running `quire preview --verbose`, you should see no errors in red, and the browser preview should work as expected.

Note that the `windows-build-tools` are part of our current Windows install instructions, but for some reason weren't working before we ran the above process again. This was likely due to some security-related issues (#16) Sophie had run into previously.
